crates/bevy_ecs/src/change_detection.rs

```diff
@@ -173,6 +173,17 @@ macro_rules! impl_into_inner {
                 self.set_changed();
                 self.value
             }
+
+            /// Maps to an inner value by applying a function to the contained reference, without flagging a change.
+            ///
+            /// You must *not* modify the argument passed to the closure.
+            /// Violating this rule will likely result in logic errors, but it will not cause undefined behavior.
+            pub fn map_unchanged<U: ?Sized>(self, f: impl FnOnce(&mut $target) -> &mut U) -> Mut<'a, U> {
```

Comment:
This sort of functional code can be hard to follow; can you add a doc test to demonstrate its usage and properties?

Reply:
What do you think of the example I used in the PR desc?

Reply:
Quite good!